docs/contributing/setting-up-project.mdx

```diff
@@ -0,0 +1,78 @@
+---
+title: Setting up the Project for Contributors
+---
+Tach CLI is an open source 
+
+#### 1. Fork the repository
+
+#### 2. Clone the repository locally
+```bash 
+git clone https://github.com/github_username/tach.git
+cd tach
+```
+It is adviced to work on a branch of your own and then raise a PR
+```bash
+git checkout -b "your_feature_name" 
+```
+
+#### 3. Create a virtual environment
+```bash
+python3 -m venv env
+```
+Activate the environment 
+```bash
+source env/Scripts/activate
+```
+
+#### 4. Install Dependencies
+```bash
+pip install -r dev-requirements.txt
+```
+
+#### 5. Build
+
+This build will reflect changes that are currently existing locally
+
+To build the project - run:
+
+```bash
+ pip install .
+```
+Note: tach is a CI tool to enforce dependencies, using modular architecture. For testing your changes you'll need to set this project up within your own project
```

Comment:
We should recommend that contributors start by running: `make deps`

This will automatically create and activate a virtual environment using `uv`, as well as build and install the project.

To rebuild after changes to Rust files, we should recommend using `make install`

See here: https://github.com/gauge-sh/tach/blob/main/Makefile